Bootbanks module fails with AttributeError

When I run acquire on ESXI I get:

```
[2024-11-12 14:56:06,669] [ERROR] Error while running module Bootbanks
Traceback (most recent call last):
  File "/opt/acquire-api/lib/python3.9/site-packages/acquire/acquire.py", line 939, in acquire_target
  File "/opt/acquire-api/lib/python3.9/site-packages/acquire/acquire.py", line 181, in run
  File "/opt/acquire-api/lib/python3.9/site-packages/acquire/acquire.py", line 690, in _run
AttributeError: 'RootFilesystemEntry' object has no attribute 'top'
[2024-11-12 14:56:06,669] [INFO ] 
```
I have a `/bootbank` entry which is symlinked to ` /vmfs/volumes/ce05e607-4f4bd1b5-8f8d-b2bfbc2c924e`

`/vmfs/volumes/ce05e607-4f4bd1b5-8f8d-b2bfbc2c924e` appears to be a root file system entry:

```
[root@localhost:/vmfs/volumes/ce05e607-4f4bd1b5-8f8d-b2bfbc2c924e] df 
Filesystem       Bytes       Used   Available Use% Mounted on
VMFS-L     24427626496 8870952960 15556673536  36% /vmfs/volumes/OSDATA-672b6bed-49e3885b-4da5-525400de4472
vfat        1073577984  211714048   861863936  20% /vmfs/volumes/BOOTBANK1
vfat        1073577984      32768  1073545216   0% /vmfs/volumes/BOOTBANK2
```

And assuming that `/vmfs/volumes/OSDATA-672b6bed-49e3885b-4da5-525400de4472`  is represented by a `RootFileSystemEntry`, something probably broke at some point, my guess would be https://github.com/fox-it/dissect.target/issues/575

--- a/acquire/acquire.py
+++ b/acquire/acquire.py
@@ -1649,7 +1649,9 @@ class Bootbanks(Module):
 
         for _, mountpoint, uuid, _ in iter_esxi_filesystems(target):
             for bootbank_path, boot_vol in boot_fs:
-                if bootbank_path.samefile(mountpoint):
+                # samefile fails on python 3.9 (https://github.com/fox-it/dissect.target/issues/1289)
+                # but support for 3.9 gets dropped soon
+                if bootbank_path.samefile(target.fs.path(mountpoint)):
                     log.info("Acquiring %s (%s)", mountpoint, boot_vol)
                     mountpoint_len = len(mountpoint)
                     base = f"fs/{uuid}:{boot_vol}"

--- a/acquire/acquire.py
+++ b/acquire/acquire.py
@@ -1639,24 +1639,24 @@ class Bootbanks(Module):
             "bootbank": "BOOTBANK1",
             "altbootbank": "BOOTBANK2",
         }
-        boot_fs = {}
+        boot_fs = []  # List of tuples of bootbank paths and volume names
 
         for boot_dir, boot_vol in boot_dirs.items():
             dir_path = target.fs.path(boot_dir)
             if dir_path.is_symlink() and dir_path.exists():
                 dst = dir_path.readlink()
-                fs = dst.get().entries.top.fs
-                boot_fs[fs] = boot_vol
+                boot_fs.append((dst, boot_vol))
 
-        for fs, mountpoint, uuid, _ in iter_esxi_filesystems(target):
-            if fs in boot_fs:
-                name = boot_fs[fs]
-                log.info("Acquiring %s (%s)", mountpoint, name)
-                mountpoint_len = len(mountpoint)
-                base = f"fs/{uuid}:{name}"
-                for path in target.fs.path(mountpoint).rglob("*"):
-                    outpath = path.as_posix()[mountpoint_len:]
-                    collector.collect_path(path, outpath=outpath, base=base)
+        for _, mountpoint, uuid, _ in iter_esxi_filesystems(target):
+            for bootbank_path, boot_vol in boot_fs:
+                if bootbank_path.samefile(mountpoint):
+                    log.info("Acquiring %s (%s)", mountpoint, boot_vol)
+                    mountpoint_len = len(mountpoint)
+                    base = f"fs/{uuid}:{boot_vol}"
+                    for path in target.fs.path(mountpoint).rglob("*"):
+                        outpath = path.as_posix()[mountpoint_len:]
+                        collector.collect_path(path, outpath=outpath, base=base)
+                    break
 
 
 @register_module("--esxi")

--- a/acquire/acquire.py
+++ b/acquire/acquire.py
@@ -1645,7 +1645,7 @@ class Bootbanks(Module):
             dir_path = target.fs.path(boot_dir)
             if dir_path.is_symlink() and dir_path.exists():
                 dst = dir_path.readlink()
-                fs = dst.get().top.fs
+                fs = dst.get().entries.top.fs
                 boot_fs[fs] = boot_vol
 
         for fs, mountpoint, uuid, _ in iter_esxi_filesystems(target):

